### DISCUSSION TITLE
Request for non-collinear functional needed in gks and spin spiral calculation( directly setting q vector)

### DISCUSSION THREAD

**Discussion by YuleZhang936** [deepmodeling/abacus-develop#5268] [General]
### Background

If abacus wants to handle non-collinear magnetism, generalized kohn-sham is needed. Thus, non-collinear functional is needed since gks needs noncollinear functional. A good application for non-collinear functional is spin spiral material.

### Describe the solution you'd like

Here is a numerically stable option for non-collinear functional. To the best of what we know, it is the only one. We hope that abacus can have this non-collinear functional approach called the multi-collinear approach.
Ref: PHYSICAL REVIEW RESEARCH 5, 013036 (2023)

### Task list only for developers

- [ ] Notice possible changes of behavior
- [ ] Explain the changes of codes in core modules of ESolver, HSolver, ElecState, Hamilt, Operator or Psi

### Task list for Issue attackers (only for developers)

- [ ] Review and understand the proposed feature and its importance.
- [ ] Research on the existing solutions and relevant research articles/resources.
- [ ] Discuss with the team to evaluate the feasibility of implementing the feature.
- [ ] Create a design document outlining the proposed solution and implementation details.
- [ ] Get feedback from the team on the design document.
- [ ] Develop the feature following the agreed design.
- [ ] Write unit tests and integration tests for the feature.
- [ ] Update the documentation to include the new feature.
- [ ] Perform code review and address any issues.
- [ ] Merge the feature into the main branch.
- [ ] Monitor for any issues or bugs reported by users after the feature is released.
- [ ] Address any issues or bugs reported by users and continuously improve the feature.

**Comment by mohanchen** [comment]
We will consider that, thanks for your advice!
